yard-mode.el

```diff
@@ -31,6 +31,22 @@
 (eval-when-compile (require 'cl))
 (require 'regexp-opt)
 
+(eval-when-compile
+  ;; add-function arrived after emacs 24.?
+  ;; old emacs try and fail to expand macros
+  (let ((nadvice-p (and (require 'nadvice nil t) (fboundp 'add-function))))
+    (defmacro yard-add-eldoc ()
```

Comment:
Why use a macro when you can just as well make that fboundp check inside a function?

Reply:
I did that originally, but the tests were failing on emacs 24.3/24.4 because of eager macro expansion introduced then.  Is there a way to avoid that?  I didn't find a solution when I searched a bit

Reply:
Or do you mean turn `yard-add-eldoc` into a function?  If so, what is the benefit?  Just in case somebody switched emacs versions and kept the same .elc files?

Reply:
Yes.

It's just simpler design.

Reply:
Ok, sure.  Quick question: does the `fbound` check need to be in an `eval-when-compile` form to avoid the eager expansion problem?  Because initially I just added those checks within the yard functions, but that was failing.  And, if so, does that mean the functions would need to be in an `eval-and-compile` form?

Reply:
Nvm, `eval-and-compile` does the trick, so the checks can just go directly in the original functions

Reply:
I don't think you need to do either of that, not even `require`, since `add-function` is autoloaded.

Reply:
@dgutov the `require` was removed, but I think the `eval-and-compile` is necessary since the tests were failing without it

Reply:
Wish I could see the test failure exactly, but OK!